Dokan/Base/BaseAPI.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  BaseAPI.swift
+//  Dokan
+//
+//  Created by Clara Mounir Adly on 22/06/2022.
+//
+
+import Foundation
+import Alamofire
+
+class BaseAPI<T:TargetType>{
+    private func buildParams(task: Task) -> ([String: Any], ParameterEncoding){
+           switch task {
+           case .requestPlain:
+               return ([:], URLEncoding.default)
+           case .requestParameters(parameters: let parameters, encoding: let encoding):
+               return (parameters, encoding)
+           }
+       }
+
+    func fetchData<M: Decodable>(target: T, responseClass: M.Type, completionHandler:@escaping (Result<M, NSError>)-> Void) {
```

Comment:
Ordering hint.. Let's keep public methods before the private ones.. It would be easier and faster to find the class scope.